src/context/WalletProvider/NativeWallet/components/LegacyTwoFactor.tsx

```diff
@@ -0,0 +1,69 @@
+import {
```

Comment:
We can have two ways of doing 2fa. This flow that you have and I think is common where a user tries to login, then if successful, they get a response for 2fa required, and get a new screen to enter 2fa.

Another approach, which is what I was thinking, was that we could have a 2fa hideable/showable input on the LegacyLogin.tsx screen that the user could choose to show or hide, they can enter 2fa there before they submit and then it would be only one call to the backend to do both login and 2fa verification. If they don't enter verification, but it's required, then they'd get a response back saying '2fa required' and then they could click the 2fa checkbox to open the 2fa input and resubmit the form.

In the latter case, this screen would not be necessary. (though I know the mockups didn't show that. We did clear that flow with product during breakdown.)

Either way should work, but the latter has a flow that could only make 1 call to the backend and then the user would leave LegacyLogin.tsx and enter NativeCreate.tsx

Reply:
Mmh, agree, which component should I use in order to toggle the 2fa input from the login modal?

Reply:
I added the `needs product` label because I think it would be simpler to get the product input directly instead of integrating something they might want to change in the future

Reply:
I would keep the stand alone 2FA screen if its required. After we get the 2FA we do the final submission. @NeOMakinG

Reply:
That makes sense to me too. @NeOMakinG lets keep it the way you have it now, we'll just make 2 api calls for users with 2fa.